include/rfl/internal/num_fields.hpp

```diff
@@ -97,20 +108,68 @@ struct CountFieldsHelper {
     }
   }
 
+  template <std::size_t size = 0>
+  static consteval std::size_t get_the_sole_nested_array_size() {
+    static_assert(size <= sizeof(T));
+    if constexpr (constructible_with_nested<0, size, 0>() &&
+                  !constructible_with_nested<0, size + 1, 0>()) {
+      return size;
+    } else {
+      return get_the_sole_nested_array_size<size + 1>();
+    }
+  }
+
+  template <std::size_t n, std::size_t total_arg_num>
+  static consteval bool has_n_base_param() {
+    if constexpr (n > total_arg_num) {
+      return false;
+    } else {
+      auto left = std::make_index_sequence<n>();
+      auto right = std::make_index_sequence<total_arg_num - n>();
+      return []<std::size_t... l, std::size_t... r>(std::index_sequence<l...>,
+                                                    std::index_sequence<r...>) {
+        return requires { T{any_base<T>(l)..., any(r)...}; };
+      }(left, right);
+    }
+  }
+
+  template <std::size_t total_arg_num, std::size_t index>
+  static consteval std::size_t base_param_num() {
+    if constexpr (!has_n_base_param<index + 1, total_arg_num>()) {
+      return index;
+    } else {
+      return base_param_num<total_arg_num, index + 1>();
+    }
+  }
+
   template <std::size_t index, std::size_t max>
-  static consteval std::size_t count_fields_impl() {
+  static consteval std::size_t constructable_no_brace_elision() {
     static_assert(index <= max);
     if constexpr (index == max) {
       return 0;
     } else {
       return 1 +
-             count_fields_impl<
+             constructable_no_brace_elision<
                  index + get_nested_array_size<index, max - index, 0>(), max>();
     }
   }
 
   static consteval std::size_t count_fields() {
-    return count_fields_impl<0, count_max_fields()>();
+    constexpr std::size_t max_fields = count_max_fields();
+    constexpr std::size_t total_args =
+        constructable_no_brace_elision<0, max_fields>();
+    constexpr std::size_t base_args = base_param_num<total_args, 0>();
+    if constexpr (total_args == 0 && base_args == 0) {
+      // Empty struct
+      return 0;
+    } else if constexpr (base_args == total_args) {
+      // Special case when the derived class is empty.
+      // In such cases the filed number is the fields in base class.
+      // Note that there should be only one base class in this case.
+      return get_the_sole_nested_array_size();
```

Comment:
This should be implemented with a recursion somehow because there are classes with multiple inheritances that can be used with structure bind. 

```cpp
struct B1 {};
struct B2 {};
struct B3 {};
struct M { int a; int b; };
struct D1 : B1, M, B3 {};
struct D2 : B2, D1, B3 {};

// ...
// D2 d;
// rfl::to_view(d) // recognized 3, but only 2 member
```